### PR TITLE
Add renaming example

### LINKED ISSUE #310: Add inventory example to implementation notes that has different existing and logical paths
Based on https://github.com/OCFL/spec/issues/275#issuecomment-463774897 it would make things clearer to have an example that has different existing and logical paths

### PR DESCRIPTION
**Pull request by zimeon**
Fixes #310. Adds "5.3 File Renaming in an OCFL Object".

### CODE REVIEW

File: draft/spec/index.html [hunk 1]
```diff
@@ -1031,6 +1031,43 @@ <h2>Versioned OCFL Object</h2>
 </pre>
     </section>
 
+    <section id="example-object-with-file-renaming">
+        <h2>File Renaming in an OCFL Object</h2>
+        <p>
+            The following example OCFL Object inventory shows how exiting file paths may differ from
```
Comment (ahankinson) [inline draft/spec/index.html]: `exiting [sic] file paths` => `storage paths`

File: draft/spec/index.html [hunk 2]
```diff
@@ -1031,6 +1031,43 @@ <h2>Versioned OCFL Object</h2>
...
+            logical file paths. The example object has just one version, <code>v1</code>, which has
+            two files with logical file paths <code>.hidden.txt</code> and
+            <code>name with spaces.txt</code> as shown in the <code>state</code> block. These names
+            have been normalized to avoid spaces and leading period in the existing file paths. The
+            <code>manifest</code> block shows <code>%2Ehidden.txt</code> and
+            <code>name%20with%20spaces.txt</code> within the <code>v1/content</code> directory. This
```
Comment (ahankinson) [inline draft/spec/index.html]: Why are spaces and punctuation urlencoded?

**Comment by ahankinson** [comment]
While I think we do need a rename example, I'm not sure this particular example gives a compelling case for where you might want to do that. 

Perhaps a more straightforward example, like "changing filename between versions" would be more straightforward?

File: draft/spec/index.html (hunk 1 above)
Comment (awoods) [reply draft/spec/index.html]: My understanding is that "existing path" is now "content path".
https://github.com/OCFL/spec/issues/313

File: draft/spec/index.html [hunk 3]
```diff
@@ -1031,6 +1031,43 @@ <h2>Versioned OCFL Object</h2>
...
+            specification does not constrain the normalization or renaming strategy adopted, in
+            this example URI encoding was used but many other approaches are possible, and they may
+            or may not yield an obvious relation between exiting file paths and logical file paths.
```
Comment (awoods) [inline draft/spec/index.html]: typo: "exiting" ..should likely be "content paths"

File: draft/spec/index.html [hunk 4]
```diff
@@ -1031,6 +1031,43 @@ <h2>Versioned OCFL Object</h2>
...
+        </p>
+<pre>
+{
+  "digestAlgorithm": "sha512",
+  "head": "v1",
+  "id": "http://example.org/rename",
+  "manifest": {
+    "2deeaf...c0f": [ "v1/content/%2Ehidden.txt" ],
+    "329ebf...a6a": [ "v1/content/name%20with%20spaces.txt" ]
```
Comment (awoods) [inline draft/spec/index.html]: Are we assuming that certain OCFL clients may likely be implemented to URL-encode/decode the "content paths" when interacting with OCFL files on disk?

**Comment by awoods** [comment]
> While I think we do need a rename example, I'm not sure this particular example gives a compelling case for where you might want to do that.
> 
> Perhaps a more straightforward example, like "changing filename between versions" would be more straightforward?

I was having similar thoughts. Or having a pairtree "content path" represented by a flat "logical path".

File: draft/spec/index.html [hunk 5]
```diff
@@ -1031,6 +1031,43 @@ <h2>Versioned OCFL Object</h2>
 </pre>
     </section>
 
+    <section id="example-object-with-file-renaming">
+        <h2>File Renaming in an OCFL Object</h2>
```
Comment (rosy1280) [inline draft/spec/index.html]: Is this really file renaming?  Or is this more about file names

File: draft/spec/index.html (hunk 1 above)
Comment (zimeon) [reply draft/spec/index.html]: I thought it best to stick with the current terminology until that is changed everywhere in response to #313

File: draft/spec/index.html (hunk 5 above)
Comment (zimeon) [reply draft/spec/index.html]: Yes, I think based on feedback the term renaming title is wrong. I'll try (using the current/old terminology) "Different Existing File Paths and Logical File Paths in an OCFL Object" or (using the new terminology) "Different Storage Paths and Logical Paths in an OCFL Object".

File: draft/spec/index.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (zimeon) [reply draft/spec/index.html]: Fixed

File: draft/spec/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (zimeon) [reply draft/spec/index.html]: Fixed

File: draft/spec/index.html (hunk 4 above)
Comment (zimeon) [reply draft/spec/index.html]: Based on comments I'll pick another example. I think a degree of filename normalization or sanitization will be normal but perhaps the URI encoding case is too confusing

File: draft/spec/index.html (hunk 3 above)
Comment (zimeon) [reply draft/spec/index.html]: fixed "exiting" but planning to keep with old terminology pending fix for #313 (unless @rosy1280  gets to that before I get this PR approved ;-) )

**Comment by zimeon** [comment]
Hmm... thinking about the two suggestions so far:
  * renaming between versions seems more complex than necessary because we can have paths that differ in just one version
  * a pairtree storage path doesn't make much sense unless the storage filename is also a hash (i.e. randomly distributed), in that case just using hashed filenames would be enough to show the example

so.. maybe the simplest example which gets the point across without focusing on the particular motivation is just to use a hash-based renaming?

**Comment by zimeon** [comment]
OK my friends, have at it again... Remember that I am intentionally sticking with the current/old terminology of logical/existing files paths pending work on a PR for  #313

**Comment by ahankinson** [comment]
It's better, and I have no *technical* reason why it shouldn't be used, so I've approved it.

However, it has caused me to think that this same mechanism might be used in anger, to facilitate filename obfuscation currently practiced in some software, e.g., SHA1 file renaming. I would be sad if our manifest / state mechanisms were used as a way of maintaining this practice, because it makes the actual digital content opaque to humans without the inventory file.

So I guess my only reservation with this example now is that I don't think it exemplifies "best practice" for making your content human-readable on-disk. I don't think we'll stop people from doing this, but I'm not entirely sure we should promote it in the spec either...

**Comment by ahankinson** [comment]
@rosy1280 are you happy with the latest changes?

**Pull request merged**